pkg/compact/compact.go

```diff
@@ -326,14 +330,34 @@ func (s *Syncer) garbageCollect(ctx context.Context, resolution int64) error {
 			return retry(errors.Wrapf(err, "delete block %s from bucket", id))
 		}
 
-		// Immediately update our in-memory state so no further call to SyncMetas is needed
-		// after running garbage collection.
-		delete(s.blocks, id)
+		// Mark the blocks as to be removed from the blocks map so that
+		// it is not considered for the next round of compaction.
+		s.blocksToDelete[id] = time.Now().UTC()
```

Comment:
Something is wrong here.. We already delete the block (above), so why this is map is named `to Delete`? (:

Reply:
So I believe we are actually deleting it immediately so we don't delay anything at the moment in this PR :thinking: We might want to change this.